features/step_definitions/journey_test_steps.rb

```diff
@@ -189,11 +189,11 @@
 
 When /I click the '(.*)' link in the '(.*)' column for the supplier '(.*)'/ do |link_name, column_name, supplier_name|
   @supplierName = supplier_name
+  
+  table_row_first_cell = page.first(:css, "td.summary-item-field-first") {|elem| elem.text == supplier_name}
```

Comment:
Yeah, I think that's what's happening.  
`first` means it will only select one element, but then this condition means you'll have an empty array if the name is wrong.  

At least, that's how I understand it.